File: app/app/src/test/java/dev/steamvault/app/ui/detail/logic/DetailWordingTest.kt
package dev.steamvault.app.ui.detail.logic

import org.junit.Assert.assertEquals
import org.junit.Test

/** WP 4b.6 review fix S4: pins the branch that qualifies "Confirmed current
 * at X" when the cache behind that confirmation has since been cleared. */
class DetailWordingTest {

    @Test
    fun `never confirmed -- last_manifest_check null regardless of last_prefill_at`() {
        assertEquals(ConfirmedCurrentWording.NEVER_CONFIRMED, confirmedCurrentWording(null, null))
        assertEquals(
            ConfirmedCurrentWording.NEVER_CONFIRMED,
            confirmedCurrentWording("2026-08-01T00:00:00Z", null),
        )
    }

    @Test
    fun `normal case -- both timestamps present`() {
        assertEquals(
            ConfirmedCurrentWording.CONFIRMED,
            confirmedCurrentWording("2026-08-01T00:00:00Z", "2026-08-02T00:00:00Z"),
        )
    }

    @Test
    fun `THE FIX -- confirmed but last_prefill_at null (post-deletion shape) is qualified, not bare`() {
        // api/README.md: DELETE /v1/cache/{appid} clears last_prefill_at
        // unconditionally but deliberately leaves last_manifest_check --
        // this is the exact resulting row shape.
        assertEquals(
            ConfirmedCurrentWording.CONFIRMED_BEFORE_CACHE_CLEARED,
            confirmedCurrentWording(null, "2026-08-02T00:00:00Z"),
        )
    }
}

File: web/tests/detail-wording.test.js
/**
 * Headless tests for web/js/lib/detail-wording.js (WP 4a.4).
 *
 * Run: node --test "web/tests/*.test.js"   (see web/tests/README.md)
 */
import { test } from "node:test";
import assert from "node:assert/strict";
import { confirmedCurrentWording, CONFIRMED_CURRENT_WORDING } from "../js/lib/detail-wording.js";

test("last_manifest_check null -- NEVER_CONFIRMED, regardless of last_prefill_at", () => {
  assert.equal(confirmedCurrentWording(null, null), CONFIRMED_CURRENT_WORDING.NEVER_CONFIRMED);
  assert.equal(confirmedCurrentWording("2026-08-01T00:00:00Z", null), CONFIRMED_CURRENT_WORDING.NEVER_CONFIRMED);
});

test("both present -- CONFIRMED", () => {
  assert.equal(
    confirmedCurrentWording("2026-08-01T00:00:00Z", "2026-08-02T00:00:00Z"),
    CONFIRMED_CURRENT_WORDING.CONFIRMED,
  );
});

test("last_manifest_check present but last_prefill_at null -- the post-deletion shape survives as CONFIRMED_BEFORE_CACHE_CLEARED", () => {
  assert.equal(
    confirmedCurrentWording(null, "2026-08-02T00:00:00Z"),
    CONFIRMED_CURRENT_WORDING.CONFIRMED_BEFORE_CACHE_CLEARED,
  );
});

test("undefined is treated the same as null (== comparison, not ===)", () => {
  assert.equal(confirmedCurrentWording(undefined, undefined), CONFIRMED_CURRENT_WORDING.NEVER_CONFIRMED);
  assert.equal(
    confirmedCurrentWording(undefined, "2026-08-02T00:00:00Z"),
    CONFIRMED_CURRENT_WORDING.CONFIRMED_BEFORE_CACHE_CLEARED,
  );
});
